.github/workflows/test.yml

```diff
@@ -28,3 +29,24 @@ jobs:
     - uses: codecov/codecov-action@v3
       with:
         token: ${{ secrets.CODECOV_TOKEN }}
+
+  publish:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Set up Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: 3.x
+      - name: Checkout code
+        uses: actions/checkout@v3
+      - name: Install dependencies
+        run: pip install --upgrade build twine
+      - name: Build distributions
+        run: python -m build
+      - name: Publish to PyPI
+        if: startsWith(github.ref, 'refs/tags/')
```

Comment:
Should we maybe skip the whole publish job if it's not a version tag? Or am I misunderstanding how this will work?

Reply:
Yeah, that's what I thought. Perhaps we could imagine that in the future `test` job will be doing something different from what `publish` is doing, and then running it every time (except for uploading) would be useful.

Perhaps we merge this PR as it is now and then we can still change things later if we see the need?

Reply:
The future will probably involve the `pypa/gh-action-pypi-publish` action and a separate build step. That's something I can setup now if you want.

Reply:
If we go for `pypa/gh-action-pypi-publish` the job here will not be run ever, but I guess that's fine, it would be interesting to see how the modern approach works.

Reply:
I can set it up in this PR or a new one.

Reply:
Probably makes sense to just continue in this one. We don't need to release anything right now so no rush with merging it. Let me know if you need anything from me to set up `gh-action-pypi-publish`.

Reply:
You'd need to setup the `release` environment on the repo and then enable trusting publishing on the package on PyPI.  That part's relatively simple and can be done at anytime.